### ISSUE TITLE
[FEAT] Pause / Take a Break / Go On Holiday

### ISSUE THREAD

**Issue by pfabri** [st3v3nmw/obsidian-spaced-repetition#798]
**Is your feature request related to a problem? Please describe.**
Sometimes I'm away for a week or more when I just simply cannot fit reviewing into my schedule. It's pretty stressful to see the several dozens of cards and notes waiting to be reviewed on my return. It's a massive backlog to work through.

**Describe the solution you'd like**
It would be really nice if I could tell the plugin that I'll be away and on my return let it know that I'm back _when_ I'm back (i.e. I would not like to specify the length of my absence in advance). In the interim review queues should not be added to.

**Additional context**
I know that metadata is stored in the files themselves. Perhaps an easy solution (at least on the face of it) would be that on the user's return the plugin could just run through all the files and add the number of days of absence to all the card and note due dates. This way the user could just pick up from the same state (i.e. same number of items to review) where they were when they went on a break.

**Comment by LennyPhoenix** [comment]
Perhaps a more general "postpone" command would be useful? That way if a user returns from a break after an arbitrary time, e.g. 2 weeks, they can simply postpone their flashcards by 14 days, and all due dates will be adjusted accordingly like you describe?

**Comment by pfabri** [comment]
Yes, good point! When I gave it a second thought I came to the same conclusion, just didn‘t get a chance to add a follow-up any earlier. Thanks!Sent from my iPhoneOn 21 Nov 2023, at 20:03, Lenny Critchley ***@***.***> wrote:﻿
Perhaps a more general "postpone" command would be useful? That way if a user returns from a break after an arbitrary time, e.g. 2 weeks, they can simply postpone their flashcards by 14 days, and all due dates will be adjusted accordingly like you describe?

**Comment by nayyala** [comment]
@ronzulu This would be amazing! Especially now, I find my reviews backig up during the holidays.

Also, I was reluctant to upgrade from 1.10.1 but have finally done so. Great work!

**Comment by nayyala** [comment]
@ronzulu Maybe one simple alternative to this that could work temporarily would be to separate cards due today and cards that are overdue in each deck? Atleast for the time being, we can review the cards that are actually due today and then slowly chip away at the overdue ones after we are back so as to avoid building up further backup. Please consider doing osmething like this, the holidays are presenting a period of absence for most of us!

**Comment by LennyPhoenix** [comment]
I'd be willing to give implementing this a try, I'll see how much time I get over the next few days.

**Comment by ronzulu** [comment]
Hi @nayyala, I won't be able to work on this, hopefully @LennyPhoenix will manage!

**Comment by nayyala** [comment]
Awesome! Thank you guys so much!

**Comment by pfabri** [comment]
This is very messy. I don’t think it would be a good idea to push this into a release hastily then take it out and replace it with the originally proposed solution in the next release. Sent from my iPhoneOn 21 Dec 2023, at 22:49, nayyala ***@***.***> wrote:﻿
@ronzulu Maybe one simple alternative to this that could work temporarily would be to separate cards due today and cards that are overdue in each deck? Atleast for the time being, we can review the cards that are actually due today and then slowly chip away at the overdue ones after we are back so as to avoid building up further backup. Please consider doing osmething like this, the holidays are presenting a period of absence for most of us!

**Comment by LennyPhoenix** [comment]
Agreed, if I get around to it I will probably just implement the postpone command directly.

**Comment by nayyala** [comment]
@LennyPhoenix have you got any template code/sample already written? I would even go so far as to try and implement it in my own system soon, as I can already see the mountain of reviews piling up for the holidays

**Comment by pfabri** [comment]
I was thinking of writing a few lines of python to implement this postpone feature _externally_ for myself until someone does it in Typescript _internally_ to the plugin and the whole thing gets released. If I do it, I could share the script here if anyone needs it.

**Comment by nayyala** [comment]
@pfabri That would be amaziing! If you get it working, I would absolutely be interested

**Comment by nayyala** [comment]
@pfabri One note that I might add, I actually did attempt it yesterday without success. I think one thing I noticed was that the magic number dates of 2000-01-01 should be skipped

**Comment by LennyPhoenix** [comment]
> @LennyPhoenix have you got any template code/sample already written? I would even go so far as to try and implement it in my own system soon, as I can already see the mountain of reviews piling up for the holidays 

Sorry no, I haven't started yet

**Comment by ronzulu** [comment]
> @pfabri One note that I might add, I actually did attempt it yesterday without success. I think one thing I noticed was that the magic number dates of 2000-01-01 should be skipped

BTW the value is kept in `CardScheduleInfo.dummyDueDateForNewCard`

**Comment by pfabri** [comment]
Those who are on Windows and don't want to deal with Python I can make an executable version (.exe) available for you. Just ask for it.

**Comment by nayyala** [comment]
@pfabri is it something that could be seamlessly integrated into the plugin you think?

**Comment by nayyala** [comment]
I still have a feeling it would be easiest to implement an offset to what the app interprets as today's date. Are there multiple variables that store this value or is there one location where this can be changed? @ronzulu

**Comment by pfabri** [comment]
@nayyala 
> @pfabri is it something that could be seamlessly integrated into the plugin you think?

It wouldn't make sense to integrate it into the SpacedRepetition plugin in its current form: my script basically does search/replace operations through regular expressions looking for date metadata in the format SpacedRepetition uses. It has no knowledge of what a Vault is, does not understand tags etc. In my opinion a plugin native to Obsidian (and Spaced Repetition) should know its environment (for example to offer postponing notes/cards under one tag only). 

The script I wrote is just a quick&dirty workaround to provide the postponing action I had a need for but isn't currently supported natively.
